packages/unmock-matcher/src/matcher.ts

```diff
@@ -0,0 +1,59 @@
+import { IMock, ISerializedRequest, ISerializedResponse } from "unmock-core";
+
+import debug from "debug";
+
+const log = debug("unmock-matcher:comparator");
+
+type RequestComparator = (
+  intercepted: ISerializedRequest,
+  mock: ISerializedRequest,
+) => boolean;
+
+const requestsMatch: RequestComparator = (
+  interceptedRequest,
+  mockRequest,
+): boolean => {
+  function strictEqualMatch(property: keyof ISerializedRequest): boolean {
```

Comment:
Not sure if this is a clean pattern - could you just define `strictEqualMatch` to take additional params you need from the scope?

Reply:
Moved it out of the function, also added prelim support for regexp in the mock.